packages/core/src/page.js

```diff
@@ -48,18 +48,38 @@ export class Page {
     });
   }
 
+  mergeCookies(userPassedCookie, autoCapturedCookie) {
+    if (!autoCapturedCookie) return userPassedCookie;
+    if (userPassedCookie.length === 0) return autoCapturedCookie;
+
+    const mergedCookies = [...userPassedCookie, ...autoCapturedCookie];
+    const uniqueCookies = [];
+    const names = new Set();
+
+    for (const cookie of mergedCookies) {
+      if (!names.has(cookie.name)) {
+        uniqueCookies.push(cookie);
+        names.add(cookie.name);
+      }
+    }
```

Comment:
write a comment that user passed cookie will override similar auto captured cookies.

Reply:
Done